docs/conf.py

```diff
@@ -40,6 +40,11 @@
 
 intersphinx_mapping = {"python": ("https://docs.python.org/3", None)}
 
+# io.BytesIO is incorrectly converted to _io.BytesIO, this alias prevents that
+autodoc_type_aliases = {
+    "BytesIO": "~io.BytesIO",
```

Comment:
```suggestion
    "BytesIO": "io.BytesIO",
```
I find that the tilde isn't necessary? https://github.com/radarhere/Pillow/commit/f698cb9ee4fd2fd5d5c66cbe1ae997938fa919a8 / https://github.com/radarhere/Pillow/actions/runs/7599204419/job/20695972684

Reply:
The tilde is for consistency with other types.
With the tilde:
<img width="386" alt="image" src="https://github.com/python-pillow/Pillow/assets/3819630/86b472ae-cd43-4d0f-99e0-49f3daf62e61">

Without the tilde:
<img width="406" alt="image" src="https://github.com/python-pillow/Pillow/assets/3819630/5eb9012c-3eaa-49dd-a165-776b332a8ee3">

Compare with e.g. [`typing.BinaryIO`](https://pillow.readthedocs.io/en/stable/reference/ImageFont.html#PIL.ImageFont.FreeTypeFont):
<img width="527" alt="image" src="https://github.com/python-pillow/Pillow/assets/3819630/eaf18c13-e332-48cb-9fdb-961d565c2b7f">